### DISCUSSION TITLE
Installation not working - SQL tables not created?

### DISCUSSION THREAD

**Discussion by tfboy** [blind-coder/rcmcarddav#471] [Q&A] [answered]
Hi,
I'm having a strange issue where, after installing via composer and following the procedure, I get an error when logging in.

Here's what I have done:

- I've logged out of roundcube
- I've installed the plugin via Composer (I ran as root, I know not recommended, but roundcube was installed as root, part of an ISPconfig installation)
- I've enabled the plugin in roundcube's `config.inc.php` file
- I log into roundcube
- After typing in my 2FA token, I get an "Oops" error.
Looking at the logs, I see this entry:
```
[29-May-2024 16:49:17 +0000]: <mdjuk070> DB Error: [1146] Table 'roundcube.carddav_addressbooks' doesn't exist (SQL Query: SELECT * FROM carddav_addressbooks WHERE ((`user_id` = '3'))) in                        /usr/share/roundcube/program/lib/Roundcube/rcube_db.php on line 578 (POST /webmail/?_task=mail&_action=)
```
Looking at the database via phpmyadmin, it looks like the tables do not exist:
![image](https://github.com/mstilkerich/rcmcarddav/assets/6972726/2c3addfe-5392-46b7-8f09-31a65d046ffe)

I'm wondering if the tables in the database haven't been created. Delving into the various files, I found the `dbmigrations/0000-dbinit/mysql.sql` but I think that syntax is wrong with the TABLE_PREFIX.

Looking at the mysql.sql file, it has entries like
```
-- table to store the configured address books
CREATE TABLE IF NOT EXISTS TABLE_PREFIXcarddav_addressbooks (
        id INT UNSIGNED NOT NULL AUTO_INCREMENT,
.
.
.
) CHARACTER SET utf8 COLLATE utf8_unicode_ci /*!40000 ENGINE=INNODB */;

CREATE TABLE IF NOT EXISTS TABLE_PREFIXcarddav_contacts (
        id INT UNSIGNED NOT NULL AUTO_INCREMENT,
        abook_id INT UNSIGNED NOT NULL,
.
.
```
Should it be `TABLE_PREFIXcarddav_contacts` ? I guess so as that file hasn't changed in some time and I remember running carddav on a previous server with no problem. Maybe this script isn't being called up, but the install says the database tables are created on user login (and that's why no one should already be logged in when running for the first time).

Any ideas?

System info:
Debian 12.5
MySQL / MariaDB 10.11.6
Roundcube 1.6.5
Carddav 5.1.0

Thanks :)

**Comment by mstilkerich** [comment, marked as answer]
Are you using the twofactor_gauthenticator plugin?

If so see #175 for issue description and workaround.

**Reply by tfboy** [reply]
Wow. Thank you. Yes, that was it.
